theano/gpuarray/opt.py

```diff
@@ -1141,7 +1156,6 @@ def local_gpua_careduce(op, context_name, inputs, outputs):
 
             reshaped_x = x.reshape(tensor.stack(new_in_shp))
             gpu_reshaped_x = as_gpuarray_variable(reshaped_x, context_name)
-            gvar = greduce(gpu_reshaped_x)
             # We need to have the make node called, otherwise the mask can
             # be None
```

Comment:
The comment is correct, we need to keep that line. So add it back.

The problem is that the call supports_c_code need the attribute reduce_mask to be not none. And currently only a call to make_node do this. It is ugly and need refactoring, but removing this line without the refactoring isn't a good think.